app/assets/javascripts/libs/lux/lux.js

```diff
@@ -807,4 +818,18 @@ LUX = function() {
         errorHandler(e)
     })), dlog("lux.js evaluation end."), _LUX
 }();
-var LUX_t_end = Date.now();
\ No newline at end of file
+var LUX_t_end = Date.now();
+
+// This ID usually appended to the end of the lux.js as a query string when
+// using the SpeedCurve hosted version - but we have to include it here as this
+// is self hosted.
+LUX.customerid = 47044334;
```

Comment:
Always good to have questions about security 👍 

This ID is okay to be public - the Speedcurve approved method™️, which we can't use, has it appended to the end of the script URI, for example:

```html
<script src="https://cdn.speedcurve.com/js/lux.js?id=47044334" async defer crossorigin="anonymous"></script>
```

Ruby on Rails changes the file name by appending a fingerprint to it for better caching - but this means that the script can't find the ID from the URI. Adding it as a variable fixes this problem.

LUX also has an allow list of domains that it will accept requests from - any requests that aren't from `gov.uk` will be rejected. This should mean that it can't be spammed with requests.